ui/src/app/views/workspace-edit/component.ts

```diff
@@ -57,46 +66,38 @@ export class WorkspaceEditComponent implements OnInit {
         population: false,
         reviewRequested: false
       }};
-    if (this.route.routeConfig.data.adding) {
-      this.adding = true;
+    this.mode = WorkspaceEditMode.Edit;
+    if (this.route.routeConfig.data.mode) {
+      this.mode = this.route.routeConfig.data.mode;
+    }
+
+    if (this.mode === WorkspaceEditMode.Create || this.mode === WorkspaceEditMode.Clone) {
+      // There is a new workspace to be created via this flow.
+      this.accessLevel = WorkspaceAccessLevel.OWNER;
       this.errorHandlingService.retryApi(this.profileService.getMe()).subscribe(profile => {
         this.workspace.namespace = profile.freeTierBillingProjectName;
       });
-      this.accessLevel = WorkspaceAccessLevel.OWNER;
-    } else {
+    }
+    if (this.mode === WorkspaceEditMode.Edit || this.mode === WorkspaceEditMode.Clone) {
+      // There is an existing workspace referenced in this flow.
```

Comment:
I find the wording in these comments to be especially nice.

Reply:
Hah - thanks.